tests/rptest/tests/full_disk_test.py

```diff
@@ -461,3 +461,77 @@ def test_basic_usage_report(self):
             self.redpanda.rolling_restart_nodes(self.redpanda.nodes)
 
             wait_until(lambda: self.check(0.05), timeout_sec=10, backoff_sec=5)
+
+
+class DiskStatsOverrideTest(RedpandaTest):
+    @cluster(num_nodes=3)
+    def test_disk_stat_total_overrides(self):
+        if not self.test_context.globals.get('use_xfs_partitions', False):
+            self.logger.warn(
+                "skipping test: looks like we are not using separate partitions for each node"
+            )
+            return
+
+        admin = Admin(self.redpanda)
+        # current size
+        orig_stat = admin.get_disk_stat("data", self.redpanda.nodes[0])
+        # set total size to 16 KB smaller
+        admin.set_disk_stat_override("data",
+                                     self.redpanda.nodes[0],
+                                     total_bytes=orig_stat["total_bytes"] -
+                                     2**14)
+        # updated current size
+        stat = admin.get_disk_stat("data", self.redpanda.nodes[0])
+
+        # the sizes aren't exact because there are conversion that involve
+        # rounding to block size. use a 4K block threshold to test.
+        delta = abs(orig_stat["total_bytes"] - 2**14 - stat["total_bytes"])
+        assert delta <= 4096
+
+    @cluster(num_nodes=3)
+    def test_disk_stat_free_overrides(self):
+        if not self.test_context.globals.get('use_xfs_partitions', False):
+            self.logger.warn(
+                "skipping test: looks like we are not using separate partitions for each node"
+            )
+            return
+
+        admin = Admin(self.redpanda)
+        # current size
+        orig_stat = admin.get_disk_stat("data", self.redpanda.nodes[0])
+        # set total size to 16 KB smaller
+        admin.set_disk_stat_override("data",
+                                     self.redpanda.nodes[0],
+                                     free_bytes=orig_stat["total_bytes"])
+        # updated current size
+        stat = admin.get_disk_stat("data", self.redpanda.nodes[0])
+
+        # the sizes aren't exact because there are conversion that involve
+        # rounding to block size. use a 4K block threshold to test.
+        delta = abs(stat["free_bytes"] - stat["total_bytes"])
+        assert delta <= 4096
+
+    @cluster(num_nodes=3)
+    def test_disk_stat_free_delta_overrides(self):
+        if not self.test_context.globals.get('use_xfs_partitions', False):
+            self.logger.warn(
+                "skipping test: looks like we are not using separate partitions for each node"
+            )
+            return
+
```

Comment:
Should these also skip if using docker? Maybe that's the intent here already, but `RedpandaService.dedicated_nodes()` seems like it might belong here too?

Reply:
>Should these also skip if using docker? Maybe that's the intent here already

The intention isn't to skip if using docker, it's to skip the test if we are running on storage whose stats are subject to noisy neighbor workloads. In CI we run in docker, but every test gets a dedicated xfs volume, so no noisy neighbor problem. But running locally on your workstation, it's unlikely you have that setup. The dedicated node case I think is fine too (at least there redpanda _should_ be the only thing running).

I removed this check in all cases for testing the change to the total disk size--this shouldn't be changing even with noisy neighbors (unlike free-space).

Did I understand your question correctly?

Reply:
Gotcha, thanks for the explanation! Yeah I didn't realize that about docker in CI, but it makes sense